frontend/src/metabase-lib/join.ts

```diff
@@ -26,6 +40,20 @@ export function availableJoinStrategies(
   return ML.available_join_strategies(query, stageIndex);
 }
 
+export function joinStrategy(join: Join): JoinStrategy {
+  return ML.join_strategy(join);
+}
+
+export function withJoinStrategy(join: Join, strategy: JoinStrategy): Join {
+  return ML.with_join_strategy(join, strategy);
+}
+
+export function joinConditions(join: Join): FilterClause[] {
+  return ML.join_conditions(join);
+}
+
```

Comment:
The expectation is that you'll use `externalOp` to convert these to external op format -- see https://github.com/metabase/metabase/issues/31677